libs/designsystem/src/lib/components/radio/radio-group/radio-group.component.spec.ts

```diff
@@ -697,4 +704,540 @@ describe('RadioGroupComponent', () => {
+
+            spectator.setHostInput('items', items);
+            await TestHelper.waitForTimeout();
+            spectator.detectChanges();
+
+            radios = spectator.queryAll(RadioComponent);
+            radios.forEach((each) => expect(each.disabled).toBeTrue());
+          });
+        });
+      });
+
+      describe('error state when the bound form group control is required', () => {
+        beforeEach(async () => {
+          favoriteFoodControl = new FormControl(items[defaultSelectedIndex], Validators.required);
+
+          spectator = createHost(
+            `<form [formGroup]="favoriteFoodForm">
+                <kirby-radio-group [items]="items" formControlName="favoriteFood">
+                </kirby-radio-group>
+            </form>`,
+            {
+              hostProps: {
+                favoriteFoodForm: new FormGroup({
+                  favoriteFood: favoriteFoodControl,
+                }),
+                items: items,
+              },
+            }
+          );
+        });
+
+        describe('when the bound form group control is not null', () => {
+          beforeEach(async () => {
+            await setFormControlValue(items[defaultSelectedIndex]);
+          });
+
+          describe('and the bound form group control is touched', async () => {
+            beforeEach(async () => {
+              spectator.component._onRadioBlur();
+              await TestHelper.waitForTimeout();
+              spectator.detectChanges();
+            });
+
+            it('should not be in error state', async () => {
+              expect(spectator.element.classList).toContain('ng-touched');
+              expect(spectator.element.classList).not.toContain('ng-invalid');
+            });
+          });
+
+          describe('and the bound form group control is untouched', async () => {
+            beforeEach(async () => {
+              favoriteFoodControl.markAsUntouched();
+              await TestHelper.waitForTimeout();
+              spectator.detectChanges();
+            });
+
+            it('should not be in error state', async () => {
+              expect(spectator.element.classList).not.toContain('ng-touched');
+              expect(spectator.element.classList).not.toContain('ng-invalid');
+            });
+          });
+        });
+
+        describe('When the bound form group control is null', () => {
+          beforeEach(async () => {
+            await setFormControlValue(null);
+          });
+
+          describe('and the bound form group control is touched', () => {
+            beforeEach(async () => {
+              spectator.component._onRadioBlur();
+              await TestHelper.waitForTimeout();
+              spectator.detectChanges();
+            });
+
+            it('should be in error state', async () => {
+              expect(spectator.element.classList).toContain('ng-touched');
+              expect(spectator.element.classList).toContain('ng-invalid');
```

Comment:
Consider expecting the rendering of the radios instead:
```suggestion
        //TODO loop radioElements, get icon for each and assert on rendered style:
         expect(radioIcon).toHaveComputedStyle({
        'border-width': '1px',
        'border-color': getColor('danger'),
      });
```